js/getMovieList.js

```diff
@@ -71,17 +72,83 @@ function appendMovieCard(movie) {
         if (movie.Poster != "N/A") {
             movieCard.style.background = `URL(${movie.Poster}) no-repeat center center/cover`;
         } else {
             movieCard.style.background = `URL(${NO_POSTER_URL}) no-repeat center center/cover`;
         }
+        //checking every card for being favorite
+        const favoriteMoviesArray = JSON.parse(localStorage.getItem('favoriteMovies'));
+        if (favoriteMoviesArray && favoriteMoviesArray.length > 0) {
+            favoriteMoviesArray.forEach(item => {
+                if(item.Title === movie.Title) {
+                    star.classList.add('movie-card__star-active');
+                    star.innerText = '★';
+                }
+            })
+        }
     document.querySelector('#movie-list').appendChild(movieCard);
-}
+    //adding event listeners
+    movieCard.addEventListener('mouseover', (e) => {
+        if(e.target === movieCard) {
+            e.target.children[2].classList.add('movie-card__star-hover');
+        }
+    })  
+    movieCard.addEventListener('mouseleave', (e) => {
+        if(e.target === movieCard) {
+            e.target.children[2].classList.remove('movie-card__star-hover');
+        }
+    })   
+
+    star.addEventListener('click', (e) => {
+        e.preventDefault();
 
+        const movieDescr = {
+            Title: movie.Title,
+            ImdbId: movie.imdbID,
+            Year: movie.Year,
+            Poster: movie.Poster,
+            Type: movie.Type
+        }
+        let currFaveMoviesArr = JSON.parse(localStorage.getItem('favoriteMovies'));
+
+            if (e.target.classList.contains('movie-card__star-active')) {
+                e.target.classList.remove('movie-card__star-active');
+                e.target.innerText = '☆';
+                removeMovieCardFromLocalstorage(currFaveMoviesArr, movieDescr.ImdbId);
+                renderFaveMovieCard();
+                } else {
+                e.target.classList.add('movie-card__star-active');
+                e.target.innerText = '★';
+                if(!currFaveMoviesArr || currFaveMoviesArr.length < 1) {
+                    currFaveMoviesArr = [];
+                    currFaveMoviesArr.push(movieDescr);
+                    localStorage.setItem('favoriteMovies', JSON.stringify(currFaveMoviesArr));
+                    renderFaveMovieCard();
+                } else {
+                    currFaveMoviesArr.push(movieDescr);
+                    localStorage.setItem('favoriteMovies', JSON.stringify(currFaveMoviesArr));
+                    renderFaveMovieCard();
+                }
+            }
+        });
+    }
+
+function removeMovieCardFromLocalstorage(arr, Id) {
+        if(arr) {
+            for (let i=0; i < arr.length; i++) {
+                if(Id == arr[i].ImdbId) {
+                    arr.splice(i, 1);
+                   }
+                   localStorage.setItem('favoriteMovies', JSON.stringify(arr));
+                }
+                
+            }
```

Comment:
let's use `.find` in combination with `.splice` instead of the for loop?

as a rule of thumb, try to avoid for loops in the code